packages/firestore/test/unit/specs/spec_test_runner.ts

```diff
@@ -417,16 +421,9 @@ abstract class TestRunner {
   ): Persistence;
   protected abstract destroyPersistence(): Promise<void>;
 
-  async start(): Promise<void> {
-    this.connection.reset();
-    await this.persistence.start();
-    await this.localStore.start();
-    await this.remoteStore.start();
-  }
-
   async shutdown(): Promise<void> {
     await this.remoteStore.shutdown();
-    await this.persistence.shutdown(/* deleteData= */ true);
+    await (await this.persistence).shutdown(/* deleteData= */ true);
```

Comment:
Please try to find a way to make this a non-Promise again.

Reply:
Restructured to have an `init` and a `start`.